File: problem_6/solution.py
def largest_non_adj_sum(arr):
    first = 0
    next = 0
    for i in arr:
        temp = max(first, next)

        first = next + i
        next = temp

    return max(first, next)


if __name__ == '__main__':
    filename = "test_cases.txt"
    with open(filename) as file:
        content = file.readline()
        while content:
            arr, ans = content.split()
            ans = int(ans)
            arr = [int(a) for a in arr.split(',')]

            if(largest_non_adj_sum(arr) == ans):
                print("Passed")
            else:
                print("Failed")
            content = file.readline()

File: problem_6/Solution.java
import java.io.*;
import java.util.*;

public class Solution {
  public int largestNonAdjSum (int[] arr) {
    int first = 0;
    int next = 0;
    for (int i : arr) {
      int temp = Math.max(first, next);
      first = next + i;
      next = temp;
    }
    return Math.max(first, next);
  }

  public static void main(String[] args) {
    Solution s = new Solution();
    try(BufferedReader reader = new BufferedReader(new FileReader("test_cases.txt"))) {
      String line = reader.readLine();
      while (line != null) {
        String[] content = line.split("\\s");
        String[] string_arr = content[0].split(",");
        int[] arr = Arrays.stream(string_arr).mapToInt(Integer::parseInt).toArray();
        int ans = Integer.parseInt(content[1]);
        if (s.largestNonAdjSum(arr) == ans) {
          System.out.println("Passed");
        } else {
          System.out.println("Failed");
        }
        line = reader.readLine();
      }
    } catch (IOException ex) {
      System.out.println(ex);
    }
  }
}
